Mathlib/FieldTheory/Finite/GaloisField.lean

```diff
@@ -222,7 +223,45 @@ end GaloisField
 
 namespace FiniteField
 
-variable {K : Type*} [Field K] [Fintype K] {K' : Type*} [Field K'] [Fintype K']
+variable {K K' : Type*} [Field K] [Field K']
+
+section norm
+
+variable [Algebra K K'] [Finite K']
+
+theorem algebraMap_norm_eq_pow {x : K'} :
+    algebraMap K K' (Algebra.norm K x) = x ^ ((Nat.card K' - 1) / (Nat.card K - 1)) := by
+  have := Finite.of_injective_finite_range (algebraMap K K').injective
+  have := Fintype.ofFinite K
+  have := Fintype.ofFinite K'
+  simp_rw [← Fintype.card_eq_nat_card, Algebra.norm_eq_prod_automorphisms,
+    ← (bijective_frobeniusAlgEquivOfAlgebraic_pow K K').prod_comp, AlgEquiv.coe_pow,
+    coe_frobeniusAlgEquivOfAlgebraic, pow_iterate, Finset.prod_pow_eq_pow_sum,
+    Fin.sum_univ_eq_sum_range, Nat.geomSum_eq Fintype.one_lt_card, ← Module.card_eq_pow_finrank]
+
+variable (K K')
+
+theorem unitsMap_norm_surjective : Function.Surjective (Units.map <| Algebra.norm K (S := K')) :=
+  have := Finite.of_injective_finite_range (algebraMap K K').injective
+  MonoidHom.surjective_of_card_ker_le_div _ <| by
+    simp_rw [Nat.card_units]
+    classical
```

Comment:
Where is this needed? It seems strange that we need `classical` for this result.

Reply:
A DecidableEq instance is needed because the statement of [IsCyclic.card_pow_eq_one_le](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/SpecificGroups/Cyclic.html#IsCyclic.card_pow_eq_one_le) uses `Finset.filter`. One could change the Finset to a Set (and use `Nat.card`/`Set.ncard` and `Finite` instead of `Fintype`) in the statement, but it's out of scope of this PR.